account_invoice_section_sale_order/tests/test_invoice_group_by_sale_order.py

```diff
@@ -80,21 +80,25 @@ def setUpClass(cls):
     def test_create_invoice(self):
         """ Check invoice is generated  with sale order sections."""
         result = {
-            0: "".join([self.order1_p1.name, " - ", self.order1_p1.client_order_ref]),
-            1: "order 1 line 1",
-            2: "order 1 line 2",
-            3: self.order2_p1.name,
-            4: "order 2 line 1",
-            5: "order 2 line 2",
+            10: (
+                "".join([self.order1_p1.name, " - ", self.order1_p1.client_order_ref]),
+                "line_section",
+            ),
+            20: ("order 1 line 1", False),
+            30: ("order 1 line 2", False),
+            40: (self.order2_p1.name, "line_section"),
+            50: ("order 2 line 1", False),
+            60: ("order 2 line 2", False),
         }
         invoice_ids = (self.order1_p1 + self.order2_p1)._create_invoices()
         lines = (
             invoice_ids[0]
             .line_ids.sorted("sequence")
             .filtered(lambda r: not r.exclude_from_invoice_tab)
         )
-        for idx, line in enumerate(lines):
-            self.assertEqual(line.name, result[idx])
+        for line in lines:
+            self.assertEqual(line.name, result[line.sequence][0])
+            self.assertEqual(line.display_type, result[line.sequence][1])
```

Comment:
would be nice to see a test on custom grouping

Reply:
Done is separated module :wink:

Reply:
sure, but if you don't install the other module and run tests for it, you'll never know if it's broken ;)

Maybe add a TODO like "TODO: test this thing here as for now is tested on only in module xyz"